desktop-widgets/btdeviceselectiondialog.cpp

```diff
@@ -449,13 +449,28 @@ void BtDeviceSelectionDialog::deviceDiscoveryError(QBluetoothDeviceDiscoveryAgen
 }
 
 extern QString markBLEAddress(const QBluetoothDeviceInfo *device);
+extern QString btDeviceAddress(const QBluetoothDeviceInfo *device, bool ble);
 
 QString BtDeviceSelectionDialog::getSelectedDeviceAddress()
 {
-	if (selectedRemoteDeviceInfo)
-		return markBLEAddress(selectedRemoteDeviceInfo.data());
+	if (!selectedRemoteDeviceInfo)
+		return QString();
 
-	return QString();
+	int bleMode = ui->bleMode->currentIndex();
+	QBluetoothDeviceInfo *device = selectedRemoteDeviceInfo.data();
+
+	switch (bleMode) {
+	case 0:
+	default:
+		// Auto
```

Comment:
here and for the rest of the cases bellow, you can move the comments to be on the `case XX: / default:` lines or on the `return XX` line (since the `return XX` lines are not that long).

this way we can save 3 lines!